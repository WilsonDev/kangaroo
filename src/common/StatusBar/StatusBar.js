import React from 'react';

import { Button } from '@blueprintjs/core';

import './StatusBar.css';

const StatusBar = () => {
  return (
    <div className="StatusBar--container">
      <Button icon="caret-down" minimal={true}>Java</Button>
    </div>
  );
};

export default StatusBar;