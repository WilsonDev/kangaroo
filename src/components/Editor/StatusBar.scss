.StatusBar--container {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: var(--status-bar-height);
}
