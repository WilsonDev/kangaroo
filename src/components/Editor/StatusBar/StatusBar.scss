.StatusBar--container {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: var(--status-bar-height);
}

.StatusBar--show-gutter {
  margin-left: 3px;
}

.StatusBar--synchronization-icon-container {
  position: relative;
}

.StatusBar--synchronization-icon {
  color: #a7b6c2;
}

.StatusBar--synchronization-icon-status {
  position: absolute;
  width: 5px;
  height: 5px;
  border-radius: 50%;
  bottom: 0;
  right: 0;
  border: 1px solid var(--background-color-primary);
  box-sizing: content-box;

  &.local {
    background-color: #BFCCD6;
  }

  &.gist {
    background-color: #3DCC91;
  }
}

.StatusBar--language-selector {
  text-transform: capitalize;
}
