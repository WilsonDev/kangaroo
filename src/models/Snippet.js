export default class Snippet {
  constructor(args) {
    const { id, title, description, content } = args;

    this.id = id;
    this.title = title;
    this.description = description;
    this.content = content;
  }
}