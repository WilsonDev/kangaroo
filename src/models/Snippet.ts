import uuidv4 from 'uuid/v4';

import { TEXT } from './languages';

export const sourceType = {
  LOCAL: 'local',
  GIST: 'gist'
}

export interface ISnippet {
  id: number;
  title: string;
  source?: string;
  uuid?: string;
  description?: string;
  language?: string;
  content?: string;
  lastUpdated?: string;
}

/** Class representing a snippet. */
export default class Snippet implements ISnippet {
  id: number;
  title: string;
  source: string;
  uuid: string;
  description: string;
  language: string;
  content: string;
  lastUpdated: string;
  
  /**
   * Create a snippet.
   * @param {object} args
   */
  constructor(args: ISnippet) {
    const {
      id,
      title,
      source = sourceType.LOCAL,
      uuid = uuidv4(),
      description = '',
      language = TEXT,
      content = '',
      lastUpdated = new Date().toISOString()
    } = args;

    this.id = id;
    this.source = source;
    this.uuid = uuid;
    this.title = title;
    this.description = description;
    this.language = language;
    this.content = content;
    this.lastUpdated = lastUpdated;
  }
}
