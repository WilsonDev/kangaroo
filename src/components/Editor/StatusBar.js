import React from 'react';
import PropTypes from 'prop-types';
import { Tooltip, Button, HTMLSelect, Position } from '@blueprintjs/core';

import Snippet from '../../models/Snippet';
import { languages } from './languages';

import './StatusBar.scss';

const StatusBar = ({ snippet, onShowGutter, onLanguageChange }) => {
  const languageItems = Object.entries(languages).map(([key, value]) =>
    ({label: value, value: key})
  );

  return (
    <div className="StatusBar--container">
      <Tooltip content="Show gutter" position={Position.TOP}>
        <Button
          small="true"
          icon="list-detail-view"
          minimal="true"
          style={{ marginLeft: "3px" }}
          onClick={onShowGutter}
        />
      </Tooltip>

      { snippet ?
        <HTMLSelect
          value={snippet.language}
          minimal="true"
          iconProps={{icon: 'caret-down'}}
          style={{textTransform: 'capitalize'}}
          onChange={onLanguageChange}
          options={languageItems}
        /> :
        <HTMLSelect
          minimal="true"
          iconProps={{icon: 'caret-down'}}
          disabled={true}
        />
      }
    </div>
  );
};

StatusBar.propTypes = {
  snippet: PropTypes.instanceOf(Snippet),
  onShowGutter: PropTypes.func.isRequired,
  onLanguageChange: PropTypes.func
};

export default StatusBar;