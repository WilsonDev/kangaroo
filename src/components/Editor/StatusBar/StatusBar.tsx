import React, { Fragment } from 'react';
import { Tooltip, Button, HTMLSelect, Position } from '@blueprintjs/core';

import TagBar from 'components/Editor/StatusBar/TagBar/TagBar';

import Snippet from 'models/Snippet';
import { languages } from 'models/languages';

import './StatusBar.scss';

type StatusBarProps = {
  snippet: Snippet | null;
  onShowGutter: () => void;
  onTagChange: (tag: string, remove: boolean) => void;
  onLanguageChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
};

const StatusBar: React.FC<StatusBarProps> = ({
  snippet,
  onShowGutter,
  onTagChange,
  onLanguageChange,
}: StatusBarProps) => {
  const languageItems = Object.entries(languages).map(([key, value]) => ({ label: value.label, value: key }));

  return (
    <div className="StatusBar--container">
      <Tooltip content="Show gutter" position={Position.TOP}>
        <Button
          small={true}
          icon="list-detail-view"
          minimal={true}
          className="StatusBar--show-gutter"
          onClick={onShowGutter}
        />
      </Tooltip>

      {snippet ? (
        <Fragment>
          <TagBar selected={snippet.tags} onSelect={onTagChange} className="StatusBar--tag-bar" />

          <HTMLSelect
            value={snippet.language}
            minimal={true}
            iconProps={{ icon: 'caret-down' }}
            className="StatusBar--language-selector"
            onChange={onLanguageChange}
            options={languageItems}
          />
        </Fragment>
      ) : (
        <HTMLSelect minimal={true} iconProps={{ icon: 'caret-down' }} disabled={true} />
      )}
    </div>
  );
};

export default StatusBar;
