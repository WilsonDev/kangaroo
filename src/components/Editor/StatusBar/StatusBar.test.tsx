import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import StatusBar from './StatusBar';
import Snippet from 'models/Snippet';

configure({ adapter: new Adapter() });

describe('<StatusBar />', () => {
  const mockProps = {
    snippet: new Snippet({ id: 0, title: 'TEST' }),
    onShowGutter: jest.fn(),
    onTagChange: jest.fn(),
    onLanguageChange: jest.fn(),
  };

  it('render without crashing', () => {
    shallow(<StatusBar {...mockProps} />);
  });

  it('matches snapshot', () => {
    const wrapper = renderer.create(<StatusBar {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
