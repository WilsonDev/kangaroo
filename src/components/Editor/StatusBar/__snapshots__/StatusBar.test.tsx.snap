// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusBar /> matches snapshot 1`] = `
<div
  className="StatusBar--container"
>
  <span
    className="bp3-popover-wrapper"
  >
    <span
      className="bp3-popover-target"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        className="bp3-button bp3-minimal bp3-small"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        style={
          Object {
            "marginLeft": "3px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-list-detail-view"
          icon="list-detail-view"
        >
          <svg
            data-icon="list-detail-view"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              list-detail-view
            </desc>
            <path
              d="M6 9H1c-.55 0-1 .45-1 1s.45 1 1 1h5c.55 0 1-.45 1-1s-.45-1-1-1zm0 4H1c-.55 0-1 .45-1 1s.45 1 1 1h5c.55 0 1-.45 1-1s-.45-1-1-1zm9-12h-5c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1h5c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zM6 5H1c-.55 0-1 .45-1 1s.45 1 1 1h5c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h5c.55 0 1-.45 1-1s-.45-1-1-1z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
    </span>
  </span>
  <div
    className="bp3-html-select bp3-minimal"
  >
    <select
      multiple={false}
      onChange={[MockFunction]}
      style={
        Object {
          "textTransform": "capitalize",
        }
      }
      value="text"
    >
      <option
        label="Text"
        value="text"
      >
        Text
      </option>
      <option
        label="JavaScript"
        value="javascript"
      >
        JavaScript
      </option>
      <option
        label="TypeScript"
        value="typescript"
      >
        TypeScript
      </option>
      <option
        label="Java"
        value="java"
      >
        Java
      </option>
      <option
        label="XML"
        value="xml"
      >
        XML
      </option>
      <option
        label="CSS"
        value="css"
      >
        CSS
      </option>
      <option
        label="SCSS"
        value="scss"
      >
        SCSS
      </option>
      <option
        label="Markdown"
        value="markdown"
      >
        Markdown
      </option>
      <option
        label="SQL"
        value="sql"
      >
        SQL
      </option>
      <option
        label="JSON"
        value="json"
      >
        JSON
      </option>
      <option
        label="HTML"
        value="html"
      >
        HTML
      </option>
      <option
        label="C#"
        value="csharp"
      >
        C#
      </option>
    </select>
    <span
      className="bp3-icon bp3-icon-caret-down"
      icon="caret-down"
    >
      <svg
        data-icon="caret-down"
        height={16}
        viewBox="0 0 16 16"
        width={16}
      >
        <desc>
          caret-down
        </desc>
        <path
          d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
          fillRule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;
